contrib/message-logging/message-logging-parser.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Parse message log binary files.  To be used in conjunction with -logmessages."""
+
+import argparse
+import os
+import sys
+from io import BytesIO
+import json
+from pathlib import Path
+from typing import Any, List
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../test/functional'))
+
+from test_framework.messages import ser_uint256     # noqa: E402
+from test_framework.mininode import MESSAGEMAP      # noqa: E402
+from test_framework.util import assert_equal        # noqa: E402
+
+TIME_SIZE = 8
+LENGTH_SIZE = 4
+MSGTYPE_SIZE = 12
+
+# The test framework classes stores hashes as large ints in many cases.
+# There isn't a way to distinguish between a large int and a large int that is actually a blob of bytes.
+# As such, they are itemized here
+# (These can be easily found by looking for calls to deser_uint256, deser_uint256_vector, and uint256_from_str in messages.py)
+HASH_INTS = [
+    "blockhash",
+    "block_hash",
+    "hash",     # A few conflicts here
```

Comment:
conflicts with what?

Reply:
This part of the parser recognizes a uint256 by the name of the member.  It doesn't take into account the name of the class.  In some classes (in messages.py), `self.hash` refers to one of these "int-encoded" uint256s, and in others it does not.

e.g.
COutPoint, `self.hash` is an "int-encoded" uint256
CTransaction, `self.hash` is a str

This is here to justify the `isinstance(val, int)` checks in `to_jsonable` but I think it's causing more confusion than it's helping.  I'll comment this somewhere else.